sdk/attestation/azure-security-attestation/src/test/java/com/azure/security/attestation/AttestationClientTestBase.java

```diff
@@ -0,0 +1,306 @@
+
+    final ClientLogger logger = new ClientLogger(AttestationClientTestBase.class);
+
+    enum ClientTypes {
+        SHARED,
+        ISOLATED,
+        AAD,
+    }
+
+    ClientTypes classifyClient(String clientUri) {
+        assertNotNull(clientUri);
+        String regionShortName = getLocationShortName();
+        String sharedUri = "https://shared" + regionShortName + "." + regionShortName + ".test.attest.azure.net";
+        if (sharedUri.equals(clientUri)) {
+            return ClientTypes.SHARED;
+        } else if (getIsolatedUrl().equals(clientUri)) {
+            return ClientTypes.ISOLATED;
+        } else if (getAadUrl().equals(clientUri)) {
+            return ClientTypes.AAD;
+        }
+        throw new IllegalArgumentException();
+    }
+
+    InputStream base64ToStream(String base64) {
+        byte[] decoded = Base64.getDecoder().decode(base64);
+        return new ByteArrayInputStream(decoded);
+    }
+
+    AttestationClientBuilder getBuilder(HttpClient httpClient, String clientUri) {
+        return new AttestationClientBuilder().pipeline(getHttpPipeline(httpClient)).instanceUrl(clientUri);
+    }
+
+    HttpPipeline getHttpPipeline(HttpClient httpClient) {
+        TokenCredential credential = null;
+
+        if (!interceptorManager.isPlaybackMode()) {
+            credential = new EnvironmentCredentialBuilder().httpClient(httpClient).build();
+        }
+
+        final List<HttpPipelinePolicy> policies = new ArrayList<>();
+        if (credential != null) {
+            policies.add(new BearerTokenAuthenticationPolicy(credential, DATAPLANE_SCOPE));
+        }
+
+        if (getTestMode() == TestMode.RECORD) {
+            policies.add(interceptorManager.getRecordPolicy());
+        }
+
+        return new HttpPipelineBuilder()
+            .policies(policies.toArray(new HttpPipelinePolicy[0]))
+            .httpClient(httpClient == null ? interceptorManager.getPlaybackClient() : httpClient)
+            .build();
+    }
+
+    Mono<JWTClaimsSet> verifyAttestationToken(HttpClient httpClient, String clientUri, String attestationToken) throws ParseException {
+        SignedJWT token = SignedJWT.parse(attestationToken);
+
+        return getSigningCertificateByKeyId(token, httpClient, clientUri)
+            .flatMap(cert -> {
+                PublicKey key = cert.getPublicKey();
+                RSAPublicKey rsaKey = (RSAPublicKey) key;
+
+                RSASSAVerifier verifier = new RSASSAVerifier(rsaKey);
+                try {
+                    assertTrue(token.verify(verifier));
+                } catch (JOSEException e) {
+                    logger.logExceptionAsError(new RuntimeException(e.toString()));
+                }
+
+
+                JWTClaimsSet claims = null;
+                try {
+                    claims = token.getJWTClaimsSet();
+                } catch (ParseException e) {
+                    logger.logExceptionAsError(new RuntimeException(e.toString()));
```

Comment:
You can use `logThrowableAsError(Throwable)` instead so you don't create a new RuntimeException. Also, you don't need to `assertNull(claims)`. You can do `return Mono.error(logger.logThrowableAsError(e));`.